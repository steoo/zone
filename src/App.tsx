import React, { FunctionComponent, useEffect } from 'react';
import { batch, useDispatch, useSelector } from 'react-redux';
import { fetchGenres } from './features/genres/Genres.slice';
import { fetchMovies, selectMovies } from './features/movies/Movies.slice';
import Movies from './features/movies/Movies.component';

/**
 * AppComponent
 *
 * This component contains the whole app
 * */
const App: FunctionComponent = () => {
  const dispatch = useDispatch();
  const movies = useSelector(selectMovies);

  useEffect(() => {
    /**
     * Bootstrap for this component fetch all you need for the whole application
     * This side-effect uses `batch` to avoid pushing two actions one after the other
     * */
    batch(() => {
      dispatch(fetchMovies());
      dispatch(fetchGenres());
    });
  }, [dispatch]);

  return (
    <div>
      <Movies movies={movies} />
    </div>
  );
};

export default App;
