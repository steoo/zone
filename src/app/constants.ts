export const API_URL = 'http://localhost:3004';
export const IMAGE_BASE_URL = 'https://image.tmdb.org/t/p/w200';
