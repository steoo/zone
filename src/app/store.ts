import { createEpicMiddleware, ActionsObservable, combineEpics, Epic, StateObservable } from 'redux-observable';
import { catchError } from 'rxjs/operators';
import { configureStore, ThunkAction, Action } from '@reduxjs/toolkit';
import counterReducer from '../features/counter/counterSlice';
import moviesReducer from '../features/movies/Movies.slice';
import { counterEpic } from '../features/counter/Counter.epics';
import { moviesEpic } from '../features/movies/Movies.epics';

const epics: Epic[] = [counterEpic, moviesEpic];

const epicMiddleware = createEpicMiddleware();

export const store = configureStore({
  reducer: {
    counter: counterReducer,
    movies: moviesReducer
  },
  middleware: [epicMiddleware]
});

const rootEpic = (action$: ActionsObservable<Action>, store$: StateObservable<any>, dependencies: any) =>
  combineEpics(...epics)(action$, store$, dependencies).pipe(
    catchError((error, source) => {
      console.error(error);
      return source;
    })
  );

epicMiddleware.run(rootEpic);

export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<ReturnType, RootState, unknown, Action<string>>;
