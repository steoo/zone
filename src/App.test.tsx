import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { store } from './app/store';
import App from './App';

describe('Components', () => {
  describe('App', () => {
    it('should render without crashing', () => {
      const { baseElement } = render(
        <Provider store={store}>
          <App />
        </Provider>
      );

      expect(baseElement).toMatchSnapshot();
    });
  });
});
