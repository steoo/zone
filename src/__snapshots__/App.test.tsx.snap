// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components App should render without crashing 1`] = `
<body>
  <div>
    <div>
      <div />
    </div>
  </div>
</body>
`;
